### ISSUE TITLE
tree shake do not work with image

### ISSUE THREAD

**Issue by githoniel** [webpack-contrib/url-loader#228]
- Operating System: Win10  10.0.19041.746
- Node Version: 14.15.1
- NPM Version: 6.14.8
- webpack Version: 5.17.0/4.46.0
- url-loader Version: 4.1.1

### Expected Behavior

reproduce repo: https://github.com/githoniel/webpack4-treeShake

I want to get webpack to compile and make tree shake for `components`, so I've add 

```
"sideEffects": [
    "components/*",
    "*.css"
  ]
```

in package.json.

I want webpack to tree shaking everything in  `components/button` when I only import `components/Menu`。

### Actual Behavior

both webpack4 and webpack5 will achieve tree shake for `components/button/index.js` and `components/button/index.css`. 

But If I use `url-loader` and image size exceeds limit, the `components\button\style\avatar.png` will not be shaked.
It can only be shake when its size does not exceeds  limit and url-loader make it inline so it can be removed with `components/button/index.css`.

it will work well using `type: 'asset/resource'` in webpack5.

### Code

see reproduce repo

### How Do We Reproduce?

https://github.com/githoniel/webpack4-treeShake

**Comment by alexander-akait** [comment]
Yep, some limitation, you need to use `asset/resource` for webpack@5

**Comment by githoniel** [comment]
thanks for explaination
